src/Metalama.Patterns.NotifyPropertyChanged/Implementation/Natural/BuildAspectContext.cs

```diff
@@ -0,0 +1,451 @@
+
+    public IMethod? BaseOnUnmonitoredObservablePropertyChangedMethod => this._baseOnUnmonitoredObservablePropertyChangedMethod.Value;
+
+    public bool HasInheritedOnChildPropertyChangedPropertyPath( string parentPropertyPath )
+    {
+        this._inheritedOnChildPropertyChangedPropertyPaths ??=
+            BuildPropertyPathLookup( GetPropertyPaths( this.Elements.OnChildPropertyChangedMethodAttribute, this.BaseOnChildPropertyChangedMethod ) );
+
+        return this._inheritedOnChildPropertyChangedPropertyPaths.Contains( parentPropertyPath );
+    }
+
+    public bool HasInheritedOnUnmonitoredObservablePropertyChangedProperty( string propertyName )
+    {
+        this._inheritedOnUnmonitoredObservablePropertyChangedPropertyNames ??=
+            BuildPropertyPathLookup(
+                GetPropertyPaths( this.Elements.OnUnmonitoredObservablePropertyChangedMethodAttribute, this.BaseOnUnmonitoredObservablePropertyChangedMethod ) );
+
+        return this._inheritedOnUnmonitoredObservablePropertyChangedPropertyNames.Contains( propertyName );
+    }
+
+    public CertainDeferredDeclaration<IMethod> OnPropertyChangedMethod { get; } = new();
+
+    public CertainDeferredDeclaration<IMethod> OnChildPropertyChangedMethod { get; } = new();
+
+    public DeferredDeclaration<IMethod> OnUnmonitoredObservablePropertyChangedMethod { get; } =
+        new( willBeDefined: true ); // TODO: Decide according to configuration.
+
+    public List<string> PropertyPathsForOnChildPropertyChangedMethodAttribute { get; } = new();
+
+    public List<string> PropertyNamesForOnUnmonitoredObservablePropertyChangedMethodAttribute { get; } = new();
+
+    /// <summary>
+    /// Gets the <see cref="IProperty"/> for property <c>EqualityComparer<paramref name="type"/>>.Default</c>.
+    /// </summary>
+    /// <param name="type"></param>
+    /// <returns></returns>
+    public IProperty GetDefaultEqualityComparerForType( IType type )
+        => this.Elements.EqualityComparerOfT.WithTypeArguments( type ).Properties.Single( p => p.Name == "Default" );
+
+    private DependencyGraphNode? _dependencyGraph;
+    private bool _prepareDependencyGraphReportedErrors;
+
+    public DependencyGraphNode PrepareDependencyGraph()
+    {
+        var graph = Implementation.DependencyGraph.GetDependencyGraph<DependencyGraphNode>(
+            this.Target,
+            ( diagnostic, location ) =>
+            {
+                this._prepareDependencyGraphReportedErrors |= diagnostic.Definition.Severity == Severity.Error;
+                this.Builder.Diagnostics.Report( diagnostic, location.ToDiagnosticLocation() );
+            } );
+
+        foreach ( var node in graph.DescendantsDepthFirst() )
+        {
+            this._prepareDependencyGraphReportedErrors |= !node.Initialize( this );
+        }
+
+        return graph;
+    }
+
+    public DependencyGraphNode DependencyGraph => this._dependencyGraph ??= this.PrepareDependencyGraph();
```

Comment:
For clarity of the data flow, this property should not be "lazy".

Reply:
Most laziness has been removed, including here.